pkg/internal/enhancedmetrics/service/dynamodb/client.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2026 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+type awsClient interface {
```

Comment:
What is the purpose of a private interface?

Reply:
These interfaces are private to reduce their visibility, as they are used only for the mocks that test the clients' wrappers defined in that implementation.

Wrapped clients encapsulate logic, such as pagination, and their interfaces are narrower and more semantically meaningful in the context of the enhanced metrics. Therefore, they are used in the enhanced services' tests.